pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -490,6 +490,699 @@ bool is_same_shape(t_ndarray a, t_ndarray b)
     return (true);
 }
 
+void copy_data_from_bool(t_ndarray **ds, t_ndarray src, uint32_t offset)
+{
+    t_ndarray *dest = *ds;
+    unsigned char *d = (unsigned char*)dest->raw_data;
+    unsigned char *s = (unsigned char*)src.raw_data;
+    int64_t j = 0;
+
+    switch(dest->type)
+    {
+        case nd_int16:
+            for (int64_t i = 0; i < src.buffer_size; i += src.type_size)
+            {
+                *(int16_t *)(d + (offset * dest->type_size) + j) = *(bool *)(s + i);
+                j += dest->type_size;
+            }
+            break;
+
+        case nd_int32:
+            for (int64_t i = 0; i < src.buffer_size; i += src.type_size)
+            {
+                *(int32_t *)(d + (offset * dest->type_size) + j) = *(bool *)(s + i);
+                j += dest->type_size;
+            }
+            break;
+
+        case nd_int64:
+            for (int64_t i = 0; i < src.buffer_size; i += src.type_size)
+            {
+                *(int64_t *)(d + (offset * dest->type_size) + j) = *(bool *)(s + i);
+                j += dest->type_size;
+            }
+            break;
+
+        case nd_float:
+            for (int64_t i = 0; i < src.buffer_size; i += src.type_size)
+            {
+                *(float *)(d + (offset * dest->type_size) + j) = *(bool *)(s + i);
```

Comment:
@VaradRengaraj Is this equivalent to:
```suggestion
                dest->raw_data[offset * dest->type_size + j] = (void)src.nd_float[s+i];
```
?
Is there a reason to prefer complicated (and therefore error prone) pointer arithmetic?

Regarding your question on #1641 you need to ensure that this warning is not raised so the PR can be merged. This can usually be done with things like `(float*)` . I'm not sure why the error here is:
> Casting between unsigned char * and float * which have an incompatible binary data representation.

I would have expected
> Casting between float and bool which have an incompatible binary data representation.